### ISSUE TITLE
Error when loading MongoDB collection

### ISSUE THREAD

**Issue by sisp** [ContinuumIO/blaze#941]
Hi there,

I'm getting the error shown below when trying load a MongoDB collection.

`data = blaze.Data('mongodb://localhost/mydb::mycollection')`

I'm using the latest github version. The same command works with 0.7.0. Any ideas?

Thanks,
Sigurd

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sigurd/.local/lib/python2.7/site-packages/blaze-0.7.1-py2.7.egg/blaze/interactive.py", line 45, in Data
    columns=columns, **kwargs)
  File "/home/sigurd/.local/lib/python2.7/site-packages/into-0.1.3-py2.7.egg/into/regex.py", line 64, in __call__
    return self.dispatch(s)(s, *args, **kwargs)
TypeError: resource_mongo() takes exactly 2 arguments (1 given)
```

**Comment by mrocklin** [comment]
Yeah, I ran into this myself recently.  Hope to fix it in the next week or so.  Until then here is a work-around

    data = blaze.Data(blaze.resource('mongodb://localhost/mydb::mycollection'))

**Comment by sisp** [comment]
Is it an easy fix? I'm new to Blaze's internals, so if you could guide me through the process, I'd be happy to provide a fix.

**Comment by mrocklin** [comment]
Ah, sorry for letting this hang.  In short the solution is that we need to teach blaze+into how to interact with Mongo Databases themselves rather than just individual collections.  Generally Blaze considers a database of tables to be something like a dictionary of tables.  This lets us use field/column access to get at particular tables e.g. `db.mytable`.  I think that the solution should be the following:

1.  Add a resource function for MongoDB databases (e.g. `mongodb://localhost/mydb`) to complement the existing resource function for collections (e.g. `mongodb://localhost/mydb::mycollection`).  This should happen in [into](https://github.com/ContinuumIO/into/blob/master/into/backends/mongo.py#L81-L101).  
2.  Add a similar function for discover.  This would look something like [this equivalent function for SQL metadata objects](https://github.com/ContinuumIO/into/blob/master/into/backends/sql.py#L127-L145)
3.  Teach Blaze to translate `Field` access on Mongo DataBases into getting out the right collection.  This should look similar to [the equivalent function for SQL metadata objects](https://github.com/ContinuumIO/blaze/blob/master/blaze/compute/sql.py#L568-L570)

At the end of these three steps we should have something like the following:

```python
db = Data('mongodb://localhost/db')
db.  <tab>  
see list of collections

coll = db.mycollection  # a particular collection
```

We started doing this for SQL, HDF5 and other similar systems that had subcollections like tables and it made a lot of things much simpler.  Sadly we forgot about poor mongo in this process.

**Comment by michaelaye** [comment]
Is this the same thing? The error messages looks different:

```python
bz.Data('mongodb://localhost/planet_four::planet_four_subjects')

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-31e65e21a64a> in <module>()
----> 1 bz.Data('mongodb://localhost/planet_four::planet_four_subjects')

/Users/klay6683/miniconda3/lib/python3.4/site-packages/blaze/interactive.py in Data(data, dshape, name, fields, columns, schema, **kwargs)
     44             data, sub_uri = data.split('::')
     45         data = resource(data, schema=schema, dshape=dshape, columns=columns,
---> 46                         **kwargs)
     47     if (isinstance(data, Iterator) and
     48             not isinstance(data, tuple(not_an_iterator))):

/Users/klay6683/miniconda3/lib/python3.4/site-packages/odo/regex.py in __call__(self, s, *args, **kwargs)
     62 
     63     def __call__(self, s, *args, **kwargs):
---> 64         return self.dispatch(s)(s, *args, **kwargs)
     65 
     66     @property

TypeError: resource_mongo() missing 1 required positional argument: 'collection_name'
```

**Comment by mrocklin** [comment]
Yes, this is the same problem.

**Comment by michaelaye** [comment]
Is there no current workaround? I would like to convert a Mongo database into a pandas HDF but don't see a path using `odo` without it accepting any Mongo database URL? Or could I work around somehow using the .bson/.json files I have from the Mongo database dumps?

**Comment by mrocklin** [comment]
I don't think odo has this problem.  This is only a problem when using `Data`.  In that the case the workaround is to use Data(resource('...'))

**Comment by mrocklin** [comment]
@cpcloud is this fixed by #1013 ?

**Comment by cpcloud** [comment]
It should be. I'm afk for the rest of the night. I can check tomorrow for sure and close or fix accordingly.
